tools/gcc-bridge/runtime/src/main/java/org/renjin/gcc/runtime/Stdlib.java

```diff
@@ -251,6 +252,11 @@ public static int sprintf(BytePtr string, BytePtr format, Object... arguments) {
     return snprintf(string, Integer.MAX_VALUE, format, arguments);
   }
 
+  public static int vsnprintf(BytePtr string, int limit, BytePtr format, @VarArgs OpaquePtr<?> arguments) {
```

Comment:
The compile currently supports two stratgies to variadic functions:

[JvmVarArgsStrategy](https://github.com/bedatadriven/renjin/blob/7aedc16dab3e9dd56ecfb2c6dee88a1aa32efd8a/tools/gcc-bridge/compiler/src/main/java/org/renjin/gcc/codegen/type/JvmVarArgsStrategy.java#L47-L47)

The compiler takes any extra arguments, allocates an Object[] array for them, and passes them to the final argument of a method with the signature `Object... args`. This is what you see in the implementations of `sprintf`, etc. in this class. 

[VPtrVariadicStrategy](https://github.com/bedatadriven/renjin/blob/7aedc16dab3e9dd56ecfb2c6dee88a1aa32efd8a/tools/gcc-bridge/compiler/src/main/java/org/renjin/gcc/codegen/vptr/VPtrVariadicStrategy.java#L54-L54) 

The compiler takes any extra arguments to a function, and allocates a `VPtr` block into which their values are copied, and passes this to a final argument with the signature `@VarArgs VPtr arguments`. This more closely approximates how variadic args are implemented in the C memory model, and so we can fully support variadic functions written in C/C++.

The first strategy was really just a quick and dirty way to get `printf()` and friends working. At some point we'll need a better implementation that more closely follows the C standard for `printf.` Currently, we just pass the arguments we get to Java's `String.format()`, which doesn't support `%p` or many other format specifiers. 

I've looked at simply importing the C implementation of these functions from a C Standard Library implementation like musl, but they tend to be tightly coupled to the library's I/O implementation.  The other option is to use Renjin's [Formatter](https://github.com/bedatadriven/renjin/blob/4c9550f295bdb231bf17d6a44579127356432bf0/core/src/main/java/org/renjin/primitives/text/Formatter.java#L446-L446) class as a starting point, perhaps pulling it up into the gcc-runtime module and than extracting some kind of ArgumentAccessor interface so that it can work with input from SEXPs or from VPtrs.

In any case, the quick solution here is to simply declare `vsnprintf` as:

```.java
public static int vsnprintf(Ptr string, int limit, Ptr format, Object... arguments)
```

Please also add a test case to [varargs.c](https://github.com/bedatadriven/renjin/blob/7aedc16dab3e9dd56ecfb2c6dee88a1aa32efd8a/tools/gcc-bridge/compiler/src/test/resources/org/renjin/gcc/varargs.c#L23-L22). These tests are run by [GimpleCompilerTest.varArgCalls](https://github.com/bedatadriven/renjin/blob/068112d8eaa2b173a12b7fec5a160bd09504ca8c/tools/gcc-bridge/compiler/src/test/java/org/renjin/gcc/GimpleCompilerTest.java#L789-L789).